internal/connector/internal/migrations/202203030000_add_connector_namespace_lease.go

```diff
@@ -0,0 +1,44 @@
+package migrations
+
+// Migrations should NEVER use types from other packages. Types can change
+// and then migrations run on a _new_ database will fail or behave unexpectedly.
+// Instead of importing types, always re-create the type in the migration, as
+// is done here, even though the same type is defined in pkg/api
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+	"time"
+)
+
+func addConnectorNamespaceLease(migrationId string) *gormigrate.Migration {
+
+	type LeaderLease struct {
```

Comment:
What is the purpose of this `LeaderLease` entity/table?

Reply:
It's the framework bits required for background workers and reconcile threads.